crates/subspace-proof-of-time/src/pot_aes.rs

```diff
@@ -0,0 +1,238 @@
+//! AES related functionality.
+
+extern crate alloc;
+
+use aes::cipher::generic_array::GenericArray;
+use aes::cipher::{BlockEncrypt, KeyInit};
+use aes::Aes128;
+use alloc::vec::Vec;
+#[cfg(any(feature = "parallel", test))]
```

Comment:
You can run tests on `no_std` platform, you also might want to run single-threaded verification in tests, this should not be enabled in tests unconditionally (here and in all places below too):
```suggestion
#[cfg(feature = "parallel")]
```

We just need to enable both `std` and `parallel` by default as I suggested for `Cargo.toml`.

Reply:
On more thought, we want to be able to test both sequential/parallel versions, irrespective of the feature flags. Hence restored this to check for `test` attribute

Reply:
If parallel version doesn't compile, you can't test it on `no_std` anymore. Also you're testing low-level components, so single-threaded code path is still tested when you compile a parallel version.

I have not played with it yet (but plan to for blst crate), but with rayon you can actually customize the thread pool used when calling something that uses rayon, we can check that and run a different code path depending on the number of threads in that thread pool. If platform only supports one thread, compiler should be able to compile it down as if no parallel version exists.

Reply:
> If parallel version doesn't compile, you can't test it on no_std anymore. Also you're testing low-level components, so single-threaded code path is still tested when you compile a parallel version.

Didn't get this, can you explain this bit more?

Reply:
I'm basically saying that it doesn't cover all situations the way it should IMHO. But it'll probably be fine for us.